### ISSUE TITLE
[Feature request] Band limit option when identifying a cell

### ISSUE THREAD

**Issue by chau-intl** [jorgealmerio/valuetool#9]
When I hover over a cell in QGIS `value tool` will send an `identify` request to the selected raster layer. If I have a raster layer with thousands of bands this look up takes quite a while and QGIS freezes mean while.

Would it be possible some how to configure the tool to either lookup all, specific or maybe just the visible bands?

It looks like the QgsRasterDataProvider has a [sample](https://github.com/qgis/QGIS/blob/ce9edb667e819db19362cb6e05fe9431114c74c4/src/core/raster/qgsrasterdataprovider.cpp#L316) function which can be used on individual bands instead of all bands like the [identify](https://github.com/qgis/QGIS/blob/ce9edb667e819db19362cb6e05fe9431114c74c4/src/core/raster/qgsrasterdataprovider.cpp#L251) does.

**Comment by jorgealmerio** [comment]
Yes, you can!
Just choose the desired raster bands on options tab / Show Bands / Selected bands and click on the little button right side as shown in figure below.
![valueTool_bands](https://user-images.githubusercontent.com/16169268/92602437-876e9e80-f284-11ea-8a2d-3f91cca430b8.jpg)

**Comment by chau-intl** [comment]
But as far as I can tell that only limits what I see in the `table` tab and not the actual QGIS request to the raster layer. See the [getValue](https://github.com/jorgealmerio/valuetool/blob/aff96fb4a2299894f31b0d3fdf3b7ad94dacf843/core/valuewidget.py#L259) method.

So if my raster contains 3 bands and I only select one band as you have shown in the image, QGIS will return the values for all 3 bands but value tool only displays the value for the band I selected. Thus if my raster contains 1000 bands the values for all 1000 bands will be fetched but value tool only displays what I configured the layer to show.

Does that make sense?

**Comment by jorgealmerio** [comment]
You are right, the tool algorithm could be optimized to fetch only what it requested by user.
Maybe I improve it someday.
Thank you for the comments!
And sorry for my delayed answer!

**Comment by jfbourdon** [comment]
Should be adressed now, but one should use QGIS 3.22 because of a bug in previous version of QGIS with the sampling method I used.
